File: language-code/doc-kotlin/Classes and ObjectsInterfaces_7_e784ec1ef9974817bbb4d6bebb409818.kt
interface Named {
    val name: String
}
​
interface Person : Named {
    val firstName: String
    val lastName: String
    
    override val name: String get() = "$firstName $lastName"
}
​
data class Employee(
    // implementing 'name' is not required
    override val firstName: String,
    override val lastName: String,
    val position: Position
) : Person

File: language-code/doc-kotlin/Classes and ObjectsInterfaces_7_e784ec1ef9974817bbb4d6bebb409818.swift
import Foundation

protocol Named {
    var name: String { get }
}

protocol Person: Named {
    var firstName: String { get }
    var lastName: String { get }
}

extension Person {
    var name: String {
        get { "\(firstName) \(lastName)" }
    }
}

class Employee: Person {
    let firstName: String
    let lastName: String
    let position: Position
    
    init(firstName: String, lastName: String) {
        self.firstName = firstName
        self.lastName = lastName
    }
}
